pkg/controller/migmigration/directimagemigration.go

```diff
@@ -41,6 +42,28 @@ func (t *Task) getDirectImageMigration() (*migapi.DirectImageMigration, error) {
 	return nil, nil
 }
 
+func (t *Task) getDirectImageStreamMigrationForDIM(dim *migapi.DirectImageMigration) (*migapi.DirectImageStreamMigration, error) {
+
+	dimLabels := dim.GetCorrelationLabels()
+	selector := labels.SelectorFromSet(dimLabels)
+	dismList := migapi.DirectImageStreamMigrationList{}
+	err := t.Client.List(context.TODO(),
+		&k8sclient.ListOptions{
+			LabelSelector: selector,
+		},
+		&dismList)
+
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+
+	if len(dismList.Items) > 0 {
+		return &dismList.Items[0], nil
```

Comment:
Shouldn't we be cleaning up all DISM resources, not just the first? There will be one for every imagestream in the migrated namespaces.

Reply:
Updated the delete logic.